department-of-veterans-affairs/caseflow
# APPEALS-30941 ‘Personnel’ condition for report

Resolves [APPEALS-30941: ‘Personnel’ condition for report](https://jira.devops.va.gov/browse/APPEALS-30941)

# Description
Added a dropdown for the Personnel component

## Acceptance Criteria
- [x] If user selects ‘Personnel’ condition, VHA Team Members selection drop-down appears. 
- [x] User can type to find Team member name or select from drop-down.
- [x] Drop-down options for ‘VHA Team members’ include all Caseflow users in VHA business line.
- [x] User can type to search or select more options from the drop-down selection to include all desired options.
- [x] User can remove any selected Personnel displayed via the 'X' next to each selection. 
- [x] If Personnel filter condition is added, Facility filter condition is unable to be selected for this report.
- [x] At least one person must be selected if condition is added

## Testing Plan
1. run `make client-install` in the console
2. Login as a VHA admin user
3. Navigate to /decision/vha/report
4. Click on the Report Type drop-down, and select "Event / Action"
5. Click the "Add Condition" button
6. Click on the Variable drop-down, and select "Personnel"
7. Click on the Variable drop-down, and verify that "Facility" is not an option
8. Click on the "Generate task report" button and verify that an error message appears
9. Click on the VHA team members drop-down, and select one of the options
10. Repeat Step 9 and verify that the selection was added to the field 

# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---
<img width="1335" alt="Before" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110493538/f202f837-786c-43ea-a73e-d0c5d61a9ac2">|<img width="1236" alt="After_1" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110493538/6efc4d9b-9d1c-43d3-abd7-e0793c94e8ac"><img width="1242" alt="After_2" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110493538/544f866b-8992-47f2-a646-3fedcca13a2e">


# Best practices
## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?
